src/routes/update-did.ts

```diff
@@ -9,28 +9,35 @@ export class UpdateDid {
 
     public routes(app): void {
 
-        app.post('/polygon/update-did', async (req, res) => {
+        app.patch('/update-did/:did/:privateKey', async (req, res) => {
 
             try {
-                const did = req.body.did;
-                const didDoc = req.body.didDoc;
-                const privateKey = req.body.privateKey;
+                const did = req.params.did;
+                const privateKey = req.params.privateKey;
+                const didDoc = JSON.stringify(req.body.didDoc);
 
                 const updateDidRes = await updateDidDoc(did, didDoc, privateKey)
                     .then((response) => {
                         return response;
                     });
 
-                res.send(updateDidRes);
+                const gasPrice = updateDidRes.data.gasPrice;
+                const gasLimit = updateDidRes.data.gasLimit;
+
+                const gasPriceDecimal = parseInt(gasPrice._hex.substr(2), 16);
+                const gasLimitDecimal = parseInt(gasLimit._hex.substr(2), 16);
+
+                const txnFee = (gasPriceDecimal * gasLimitDecimal / Math.pow(10, 18))
```

Comment:
Can we not get the `txnFee` directly from the `txnHash`, instead of calculating it manually ?

Reply:
> Can we not get the `txnFee` directly from the `txnHash`, instead of calculating it manually ?

Sir from transaction hash we are getting the gas price and the gas limit. Multiplying both we get the txn fee. We cannot get it directly from the transaction hash received.